enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,891 @@
+need tools that make the process as simple as possible, ideally requiring no
+OpenShift knowledge.
+
+When OpenShift knowledge is required the technician performing the upgrade will
+have the support of the team of engineers that planned and vetted the upgrade.
+
+### User Stories
+
+#### Pre-load and pin upgrade images
+
+As an engineer managing a cluster that has a low bandwidth and/or unreliable
+connection to an image registry server I want to pin and pre-load all the
+images required for the upgrade so that when I decide to actually perform the
+upgrade there will be no need to contact that slow and/or unreliable registry
+server.
+
+#### Pre-load and pin custom images
+
+As an engineer managing a cluster I want to be able to pin and pre-load custom
+images required to upgrade my own applications.
+
+#### Prepare an upgrade bundle
+
+As an engineer managing a cluster I want to be able to assemble an upgrade
+bundle that contains all the artifacts (container images and metadata) needed
+to upgrade the OpenShift cluster and my own applications. I want to hand over
+this upgrade bundle and the documentation explaining how to use it to the
+technicians that will perform the upgrade in a suitable media, for example a
+USB stick.
+
+#### Include custom images in the upgrade bundle
+
+As an engineer managing a cluster I want to be able to include in the upgrade
+bundle the images required to upgrade my own workloads.
+
+#### Explicitly allow vetted upgrade bundle
+
+As an engineer managing a cluster I want be able to explicitly approve the use
+of an upgrade bundle, so that only the bundle that I tested and vetted will be
+applied to the cluster.
+
+#### Upgrade a single-node or multi-node clusters using a bundle
+
+As a technician with little or no OpenShift knowledge I want to be able to
+upgrade a single-node or multi-node cluster using the upgrade bundle and its
+documentation. I can't bring up any additional infrastructure at the cluster
+site, in particular I can't bring up an image registry server, neither outside
+of the cluster nor inside. I want to plug the USB stick provided by the
+engineer in one of the nodes of the cluster and have the rest of the process
+performed automatically.
+
+### Goals
+
+Provide an automated and documented mechanism that engineers managing a cluster
+can use to pin and pre-load images in order to upgrade a cluster without
+requiring a registry server.
+
+### Non-Goals
+
+It is not a goal to not require a registry server for other operations. For
+example, installing a new workload will still require a registry server.
+
+## Proposal
+
+### Workflow Description
+
+For all kinds of clusters, connected or disconnected, with or without an
+available registry server:
+
+1. The administrator of a cluster uses the OpenShift API to request an upgrade.
+
+1. The upgrade infrastructure of the cluster ensures that all the images
+required for the upgrade are pinned and pre-loaded in all the nodes of the
+cluster.
```

Comment:
at what point do these images get unpinned?  presumably we don't want to keep them around forever, so is there a plan to remove the N-1 images from the previous upgrade, or something along those lines?  (or at least unpin them so GC can remove them as needed)

Reply:
can you update the EP to state this?

presumably "pinned til the next upgrade" means "until the next upgrade is complete" (so we can be sure nothing is referencing the previous SHAs)